tests/include/test/psa_exercise_key.h

```diff
@@ -66,11 +66,11 @@
  * exercise_signature_key() because Mbed TLS doesn't support RIPEMD160
  * in RSA PKCS#1v1.5 signatures. A RIPEMD160-only configuration would be
  * implausible anyway. */
-#elif defined(MBEDTLS_SHA1_C)
+#elif defined(MBEDTLS_SHA1_C) || ( defined(MBEDTLS_PSA_CRYPTO_C) && defined(PSA_WANT_ALG_SHA_1) )
```

Comment:
This doesn't really match the comment "known to be supported by Mbed TLS APIs". However, I think the motivation for this is outdated: "the hashing is completed through an Mbed TLS hash API, not the PSA one" should no longer be the case. So, I suggest we rename this macro to `KNOWN_SUPPORTED_HASH_ALG` without any reference to Mbed TLS, and use purely `PSA_WANT_ALG_xxx` macros here. (And for MD5 too.)

Reply:
Yes, this needs more analysis. But I'm not sure if I correctly understand your idea with renaming `KNOWN_MBEDTLS_SUPPORTED_HASH_ALG -> KNOWN_SUPPORTED_HASH_ALG` as `KNOWN_SUPPORTED_HASH_ALG` is already provided above.

Reply:
> `KNOWN_SUPPORTED_HASH_ALG` is already provided above.

Ah that's a good point, I completely missed that :) So, perhaps my first impression was wrong and there is actually a reason that we want something supported by the Mbed TLS API... Needs more analysis indeed. Whatever the conclusion, we don't want to leave things in a state where the code does something different from what the comment says.

Perhaps we want to use `VIA_MD_OR_PSA` test macros here, and rename `KNOWN_MBEDTLS_SUPPORTED` to `KNOWN_MD_OR_PSA_SUPPORTED`?

Reply:
Or perhaps the only place where this is used should use a different macro depending on the signature algorithm used: 
* `KNOWN_SUPPORTED` for RSA PCKS#1 v1.5 and non-deterministic ECDSA,
*  `KNOWN_MD_OR_PSA_SUPPORTED` for RSA PKCS#1 v2.1
*  `KNOWN_MBEDTLS_SUPPORTED` for deterministic ECDSA (which currently depends on MD)

Reply:
I missed your previous comment. Simple switch to `KNOWN_SUPPORTED_HASH_ALG` is not enough. Let's see CI results.

Reply:
“find a supported hash, any hash” has come up several times in tests and should probably be defined in `test/psa_crypto_helpers.h`.

Reply:
API ABI check has failed otherwise it's green. But I'm not sure if current version can be accepted.

If this is fine, then `KNOWN_MBEDTLS_SUPPORTED_HASH_ALG` is not used anywhere and can be removed.

`KNOWN_SUPPORTED_HASH_ALG` is used now in `psa_excercise_key.c` (updated code snippet in previous comment) and in the following places:
https://github.com/Mbed-TLS/mbedtls/blob/149c1516e78dc84f6493cef52e4ebd615c3ba5a9/tests/include/test/psa_exercise_key.h#L94-L112

https://github.com/Mbed-TLS/mbedtls/blob/149c1516e78dc84f6493cef52e4ebd615c3ba5a9/tests/suites/test_suite_psa_crypto.function#L2612-L2616

Reply:
The current version looks good to me. Actually, removing `KNOWN_MBEDTLS_SUPPORTED_HASH_ALG` was the first thing I thought when you told me about the existence of `KNOWN_SUPPORTED_HASH_ALG` but then I thought maybe it wasn't that simple... turns out it is, so let do it!